### PR TITLE
Split section wallet

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

File: src/actor.rs [hunk 1]
```diff
@@ -423,18 +447,18 @@ impl<V: ReplicaValidator, S: Signing> Actor<V, S> {
...
             }
             ActorEvent::TransferValidationReceived(e) => {
-                if e.proof.is_some() {
-                    // if we have a proof, then we have a valid set of replicas (potentially new) to update with
-                    self.replicas = e.validation.replicas.clone();
-                }
+                // if e.proof.is_some() {
+                //     // if we have a proof, then we have a valid set of replicas (potentially new) to update with
+                //     self.replicas = e.validation.replicas.clone();
+                // }
```
Comment (Yoga07) [inline src/actor.rs:461]: Can be removed

File: src/error.rs [hunk 2]
```diff
@@ -50,14 +50,19 @@ pub enum Error {
...
     #[error("Operation out of order debit counter should be 0")]
     ShouldBeInitialOperation,
     /// No credits or debits were found to sync
-    #[error("No credits or debits found to sync to actor")]
-    NothingToSync,
+    #[error("There was no valid history found for provided actor.")]
+    NoActorHistory,
+    /// Credit history for the actor was invalid
+    #[error("Provided actor history could not be validated.")]
+    InvalidActorHistory,
```
Comment (Yoga07) [inline src/error.rs:65]: Is it meant only for Credit History? If not, we should update the doc.

**Review by Yoga07 — DISMISSED** [review]

File: src/error.rs (hunk 2 above)
Comment (joshuef) [reply src/error.rs:65]: I'll update that aye

File: src/actor.rs [hunk 3]
```diff
@@ -76,19 +77,29 @@ impl<V: ReplicaValidator, S: Signing> Actor<V, S> {
...
     ///
-    pub fn from_info(signing: S, info: WalletInfo, replica_validator: V) -> Result<Actor<V, S>> {
+    pub fn from_info(signing: S, info: WalletHistory, replica_validator: V) -> Result<Actor<V, S>> {
         let mut actor = Self::new(signing, info.replicas, replica_validator);
-        if let Some(e) = actor.from_history(info.history)? {
-            actor.apply(ActorEvent::TransfersSynched(e))?;
+        match actor.from_history(info.history) {
+            Ok(Some(event)) => actor.apply(ActorEvent::TransfersSynched(event))?,
+            Ok(None) => {}
+            Err(error) => {
```
Comment (Yoga07) [inline src/actor.rs:85]:
```suggestion
            Err(Error::NoActorHistory) => {}
            Err(e) => return Err(e)
```

**Review by Yoga07 — APPROVED** [review]

**Pull request merged**